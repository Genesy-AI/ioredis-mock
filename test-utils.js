// Runs both a regular string suite and a buffer suite
export function runTwinSuite(commandName, cb) {
  const commands = [`${commandName}Buffer`, commandName]

  return commands.forEach(command =>
    command === commandName
      ? cb(command, (a, b) => a === b)
      : cb(
          command,
          (a, b) => a.equals(Buffer.from(b))
          /*
          // @TODO: detect when it happens so we can remove this
          if (Buffer.isBuffer(a)) return a.equals(Buffer.from(b))

          console.warn('a was expected to be a Buffer:', typeof a, { a, b })
          return a === b
          // */
        )
  )
}

// Util for calling describe.skip if the suite asserts Buffer outputs, which will fail in test:browser.js due to the polyfill
export const browserSafeDescribe = command =>
  process.env.IS_BROWSER && command.endsWith('Buffer')
    ? // eslint-disable-next-line no-undef
      describe.skip
    : // eslint-disable-next-line no-undef
      describe
