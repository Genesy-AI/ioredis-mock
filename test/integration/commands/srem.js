import Redis from 'ioredis'

// eslint-disable-next-line import/no-relative-parent-imports
import { browserSafeDescribe, runTwinSuite } from '../../../test-utils'

runTwinSuite('srem', command => {
  browserSafeDescribe(command)(command, () => {
    const redis = new Redis()

    beforeEach(async () => {
      await redis.sadd('foos', 'bar', 'foo', 'baz', 'bazooka')
    })

    afterAll(() => {
      redis.disconnect()
    })

    it('should remove 1 item from set', async () => {
      expect.assertions(2)

      const status = await redis[command]('foos', 'bar')
      expect(status).toBe(1)

      const results = await redis.smembers('foos')
      expect(results).not.toContain('bar')
    })

    it('should remove 2 items from set', async () => {
      expect.assertions(3)

      const status = await redis[command]('foos', 'foo', 'baz', 'none existent')
      expect(status).toBe(2)

      const results = await redis.smembers('foos')
      expect(results).not.toContain('foo')
      expect(results).not.toContain('baz')
    })

    it("should return 0 if source don't exists", async () => {
      expect.assertions(1)

      const status = await redis[command]('bars', 'foo')
      expect(status).toBe(0)
    })

    it('should throw an exception if the key contains something other than a set', async () => {
      expect.assertions(1)
      await redis.set('foo', 'not a set')

      try {
        await redis[command]('foo', 'bar')
      } catch (err) {
        expect(err.message).toMatchSnapshot()
      }
    })
  })
})
