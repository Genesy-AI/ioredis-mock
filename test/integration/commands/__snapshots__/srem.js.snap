// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`srem should throw an exception if the key contains something other than a set 1`] = `"WRONGTYPE Operation against a key holding the wrong kind of value"`;

exports[`sremBuffer should throw an exception if the key contains something other than a set 1`] = `"WRONGTYPE Operation against a key holding the wrong kind of value"`;
